presto-client/src/main/java/com/facebook/presto/client/Column.java

```diff
@@ -28,26 +29,33 @@
     private final String name;
     private final String type;
     private final ClientTypeSignature typeSignature;
+    private final TypeMetadata typeMetadata;
 
     public Column(String name, Type type)
     {
-        this(name, type.getTypeSignature());
+        this(
+                name,
+                type.getTypeSignature().toString(),
+                new ClientTypeSignature(type.getTypeSignature()),
+                type.getTypeMetadata());
     }
 
     public Column(String name, TypeSignature signature)
```

Comment:
I think this is what you can do: remove this constructor (`Column(String name, TypeSignature signature)`). There's no real use case for this one. Create a `Type` class in client (I think call it `Type` is fine, but if it feels too confusing we can call it `ClientType`), which wraps the string type, client type signature and enum maps.